tests/ga/test_update.py

```diff
@@ -1901,5 +1892,40 @@ def time(self):
         return current_time
 
 
+class TestProcessGoalState(AgentTestCase):
```

Comment:
Test for new behaviour: now the check for new goal state is done in the UpdateHandler